File: kotlin_android_dev/app/src/main/kotlin/com/kotlin_android_dev/model/example/example_wrapper/ExampleWrapper.kt
package com.kotlin_android_dev.model.example.example_wrapper

import io.github.antonpichka.lamm.base_model.BaseModelWrapper
import io.github.antonpichka.lamm.base_model.BaseModel
import com.kotlin_android_dev.model.example.Example

open class ExampleWrapper(listObject: List<Any>) : BaseModelWrapper(listObject) {
    @Suppress("UNCHECKED_CAST")
    override fun<T : BaseModel> createModel(): T {
        return Example(listObject[0] as String) as T
    }
}

File: dart_web_dev/app/lib/model/example/example_wrapper/example_wrapper.dart
import 'package:dart_web_dev/model/example/example.dart';
import 'package:library_architecture_mvvm_modify/library_architecture_mvvm_modify.dart';
import 'package:meta/meta.dart';

@immutable
base class ExampleWrapper extends BaseModelWrapper {
  const ExampleWrapper({required List<dynamic> listObject}) : super(listObject);

  @override
  Example createModel() {
    return Example(uniqueId: listObject[0]);
  }
}